lib/widgets/recent_dm_conversations.dart

```diff
@@ -138,3 +149,55 @@ class RecentDmConversationsItem extends StatelessWidget {
           ]))));
   }
 }
+
+class _NewDmButton extends StatefulWidget {
+  final VoidCallback onPressed;
+
+  const _NewDmButton({required this.onPressed});
+
+  @override
+  State<_NewDmButton> createState() => _NewDmButtonState();
+}
+
+class _NewDmButtonState extends State<_NewDmButton> {
+  bool _pressed = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final fabBgColor = _pressed ? designVariables.fabBgPressed : designVariables.fabBg;
+    final fabLabelColor = _pressed ? designVariables.fabLabelPressed : designVariables.fabLabel;
+
+    return InkWell(
+      onTap: widget.onPressed,
+      onTapDown: (_) => setState(() => _pressed = true),
+      onTapUp: (_) => setState(() => _pressed = false),
+      onTapCancel: () => setState(() => _pressed = false),
+      borderRadius: BorderRadius.circular(28),
+      child: Container(
```

Comment:
Let's use the animated version `AnimatedContainer` here. The Figma design has specified the duration and curve for transitions.

Looks like it also handles the box shadow animation properly, so we can just complete the TODO from https://github.com/zulip/zulip-flutter/pull/1322#discussion_r2031996733, now that we do have a AnimatedScaleOnTap style widget.